Add new batch indexer:

jdrossl opened the pull request:
- Detects when a level descriptor has changed and looks for all descriptors that need to be indexed too

https://github.com/craftercms/craftercms/issues/3406

Depends on https://github.com/craftercms/search/pull/182

jdrossl committed:
Add new batch indexer:
- Detects when a level descriptor has changed and looks for all descriptors that need to be indexed too

sumerjabri reviewed (changes requested):
Shouldn't this just use Crafter Core to load the descriptors and index the result?

Why should this be aware of `level-descriptors`? These things are `Crafter Core` concerns.

Let's discuss.

jdrossl committed:
Use new item processor for detecting inherited content changes:
- Inherited content is now detected using queries intead of the repository
- Enable cache to improve performance when reindexing large sites

avasquez614 commented on src/main/resources/base-target.yaml:
```diff
@@ -51,6 +51,8 @@ target:
           # If XML merging should be done when indexing XML
           enabled: true
+          inherits:
+            field: inheritsFrom_smv
```
I think I rather have a `inheritsFrom` without `_smv`, and defined in the ES settings, just like `includedDescriptors`. I like having Crafter specific fields without a postfix. Unless doing this avoids having to do a re-index? But don't we need a re-index anyway? @sumerjabri let me know what you think too.

jdrossl replied:
yeah, I did it that way to avoid changing both ES & Solr schemas, that needs not only a reindex but creating a new index

avasquez614 replied:
Ok, re-creating an index is not worth it, let's go with the postfix.

jdrossl committed:
Fix for correctly following the indexing config
